auth/server/src/main/java/com/seagates3/dao/ldap/AccessKeyImpl.java

```diff
@@ -491,6 +491,59 @@ private void saveFedAccessKey(AccessKey accessKey)
       AccessKey[] accessKeyList = new AccessKey[accessKeys.size()];
       return (AccessKey[])accessKeys.toArray(accessKeyList);
     }
+    /**
+     * Below will delete expired access keys from ldap
+     */
+    @Override public void deleteExpiredKeys(User user)
+        throws DataAccessException {
+
+      String[] attrs = {LDAPUtils.ACCESS_KEY_ID, LDAPUtils.EXPIRY};
+
+      String accessKeyBaseDN =
+          String.format("%s=accesskeys,%s", LDAPUtils.ORGANIZATIONAL_UNIT_NAME,
+                        LDAPUtils.BASE_DN);
+
+      String filter = String.format("(&(%s=%s)(%s=%s))", LDAPUtils.USER_ID,
```

Comment:
can we add this in filter LDAPUtils.EXPIRY

Reply:
We have stored expiry date as string and we haven't defined any search type for expiry date attribute definition.
olcAttributeTypes: {12}( 1.3.6.1.4.99999.2.3.13 NAME ( 'exp' 'expiry' ) DESC '
 Expiry TimeStamp' SYNTAX 1.3.6.1.4.1.1466.115.121.1.24 )
Because of this, as of now we cannot put exp in filter (i.e. we cannot use comparator operator on exp)
In order to use expiry in ldapsearch  query filter, 
1. we need to modify attribute definition to add 'less than equal' or 'greater than equal' type
2.we need to check if it works well with strings i.e. if it perform correct date comparison or no